runelite-client/src/main/java/net/runelite/client/ui/components/ToggleState.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018, Chickensalad <https://github.com/rubeng123>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.components;
+
+/**
+ * Used to represent the state of the ToggleSwitch class
+ * @see ToggleSwitch
+ */
+public enum ToggleState
```

Comment:
UUuh, so enum that is only ON and OFF? this looks like boolean to me

Reply:
No, it should have 5 states. Don't know what he's doing.

Reply:
Wait, why 5?

Reply:
@deathbeam Surely using an enum here would be semantically clearer? I'm trying to model the status of a button, not a boolean condition?

Reply:
Doesn't the button have 5 states? for each animation frame?

Reply:
The button is either on or off. The animation frames are handled transparently based on these two states. Technically you _could_ say it has 5 states, but then why would we need an enum value for each one?
I'm using the enum as an argument for setting the state too, would it make sense to call Switch::setState(FRAME_3)? Imo it doesn't, you would want to set it only on or off right?

Reply:
Okay, makes sense, then just a boolean would be better instead of having ints for states.

Reply:
They're not ints anymore, they're enums now